lib/entities/workflow-definition.ts

```diff
@@ -14,14 +14,46 @@ import { wrapCollection } from '../common-utils'
 import enhanceWithMethods from '../enhance-with-methods'
 import { ActionType } from './role'
 
-/* Workflow Step */
+/* Workflow Step Action */
+export enum WorkflowStepActionType {
+  App = 'app',
+  Email = 'email',
+  Task = 'task',
+}
+export type WorkflowStepAction =
+  | WorkflowStepEmailAction
+  | WorkflowStepTaskAction
+  | WorkflowStepAppAction
+
+type EmailActionRecipient = string | Link<'User'> | Link<'Team'>
 
-export type WorkflowStepAction = {
-  action: Link<'AppAction'>
-  body: string
-  headers: Record<string, unknown>
+type WorkflowStepEmailAction = {
+  type: 'email'
+  configuration: {
+    recipients: EmailActionRecipient[]
+  }
 }
 
+type WorkflowStepTaskAction = {
+  type: 'task'
+  configuration: {
+    assignee: Link<'User'> | Link<'Team'>
+    body: string
+    dueDate?: number
+  }
+}
+
+type WorkflowStepAppAction = {
+  type: 'app'
+  appId: string
+  appActionId: string
+  configuration?: {
+    body?: Record<string, any>
+    headers?: Record<string, any>
```

Comment:
The value of the header is also a string, not any.

Reply:
done